rs/registry/admin/bin/main.rs

```diff
@@ -6250,17 +6378,44 @@ async fn update_registry_local_store(nns_urls: Vec<Url>, cmd: UpdateRegistryLoca
     eprintln!("Finished update.");
 }
 
-/// Returns a sender corresponding to a `test_sender`, or `current_sender` if
+/// Returns an identity corresponding to a `test_identity`, or `current_identity` if
 /// not set.
-fn get_test_sender_if_set(current_sender: Sender, test_sender: Option<u8>) -> Sender {
-    match test_sender {
-        None => current_sender,
-        Some(1) => Sender::from_keypair(&TEST_USER1_KEYPAIR),
-        Some(2) => Sender::from_keypair(&TEST_USER2_KEYPAIR),
-        Some(3) => Sender::from_keypair(&TEST_USER3_KEYPAIR),
-        Some(4) => Sender::from_keypair(&TEST_USER4_KEYPAIR),
+fn get_test_identity_if_set(
+    current_identity: Arc<dyn Identity>,
+    test_identity: Option<u8>,
+) -> Arc<dyn Identity> {
+    match test_identity {
+        None => current_identity,
+        Some(1) => {
+            let pem = TEST_USER1_KEYPAIR.to_pem();
+            Arc::new(
+                BasicIdentity::from_pem(pem.as_bytes())
+                    .expect("Failed to create BasicIdentity from test keypair"),
+            )
+        }
+        Some(2) => {
+            let pem = TEST_USER2_KEYPAIR.to_pem();
+            Arc::new(
+                BasicIdentity::from_pem(pem.as_bytes())
+                    .expect("Failed to create BasicIdentity from test keypair"),
+            )
+        }
+        Some(3) => {
+            let pem = TEST_USER3_KEYPAIR.to_pem();
+            Arc::new(
+                BasicIdentity::from_pem(pem.as_bytes())
+                    .expect("Failed to create BasicIdentity from test keypair"),
+            )
+        }
+        Some(4) => {
+            let pem = TEST_USER4_KEYPAIR.to_pem();
+            Arc::new(
+                BasicIdentity::from_pem(pem.as_bytes())
+                    .expect("Failed to create BasicIdentity from test keypair"),
+            )
+        }
```

Comment:
This is hyper repetitive. One way to fix:

```suggestion
    let Some(test_identity) = test_identity else {
        return current_identity;
    };

    let keypair = match test_identity {
        1 => TEST_USER1_KEYPAIR,
        etc...
        
        _ => panic!(...),
    };
    
    Arc::new(... keypair ...)
```